src/gui/vector/qgsattributesformmodel.h

```diff
@@ -0,0 +1,459 @@
+#ifndef QGSATTRIBUTESFORMMODEL_H
+#define QGSATTRIBUTESFORMMODEL_H
+
+#include "qgsattributeeditorelement.h"
+#include "qgsoptionalexpression.h"
+#include "qgsvectorlayer.h"
+
+#include <QAbstractItemModel>
+#include <QPushButton>
+
+
+class QgsAttributeFormTreeData
```

Comment:
All the enums/classes/structs inside this class were moved from `qgsattributesformproperties.h`

To keep them under a shared namespace, I created this `QgsAttributeFormTreeData` 'container' class. Not sure if it's the recommended way of doing it, though.